packages/hooks/use-draggable/src/index.ts

```diff
@@ -0,0 +1,110 @@
+import type {MoveMoveEvent, MoveResult} from "@react-aria/interactions";
+
+import {useEffect, useRef, useCallback} from "react";
+import {useMove} from "@react-aria/interactions";
+
+export interface UseDraggableProps {
+  /**
+   * Ref to the moving target DOM node.
+   */
+  targetRef?: React.RefObject<HTMLElement>;
+  /**
+   * Whether to disable the target is draggable.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether the target can overflow the viewport.
+   * @default false
+   */
+  canOverflow?: boolean;
+}
+
+/**
+ * A hook to make a target draggable.
+ * @param props UseDraggableProps
+ * @returns MoveResult for the drag DOM node.
+ */
+export function useDraggable(props: UseDraggableProps): MoveResult {
+  const {targetRef, isDisabled = false, canOverflow = false} = props;
+  const boundary = useRef({minLeft: 0, minTop: 0, maxLeft: 0, maxTop: 0});
+  let transform = {offsetX: 0, offsetY: 0};
+
+  const onMoveStart = () => {
+    const {offsetX, offsetY} = transform;
+
+    const targetRect = targetRef?.current?.getBoundingClientRect();
+    const targetLeft = targetRect?.left ?? 0;
+    const targetTop = targetRect?.top ?? 0;
+    const targetWidth = targetRect?.width ?? 0;
+    const targetHeight = targetRect?.height ?? 0;
+
+    const clientWidth = document.documentElement.clientWidth;
+    const clientHeight = document.documentElement.clientHeight;
+
+    const minLeft = -targetLeft + offsetX;
+    const minTop = -targetTop + offsetY;
+    const maxLeft = clientWidth - targetLeft - targetWidth + offsetX;
+    const maxTop = clientHeight - targetTop - targetHeight + offsetY;
+
+    boundary.current = {
+      minLeft,
+      minTop,
+      maxLeft,
+      maxTop,
+    };
+  };
+
+  const onMove = (e: MoveMoveEvent) => {
```

Comment:
Hello, should this function and the previous function, both being functions within a React hook and passed to another hook, be wrapped in `useCallback`?

Reply:
@AnYiEE Updated.